clients/drcachesim/tools/basic_counts.cpp

```diff
@@ -294,3 +312,47 @@ basic_counts_t::get_total_counts()
     }
     return total;
 }
+
+void
+basic_counts_t::compute_shard_interval_result(per_shard_t *shard, uint64_t interval_id)
+{
+    counters_t shard_total;
+    for (const auto &ctr : shard->counters) {
+        shard_total += ctr;
+    }
+    counters_t diff = shard_total;
+    if (!shard->per_interval_cumulative.empty()) {
+        diff -= shard->per_interval_cumulative.back();
+    }
+    shard->interval_ids.push_back(interval_id);
+    shard->per_interval_delta.push_back(diff);
+    shard->per_interval_cumulative.push_back(shard_total);
+}
+
+bool
+basic_counts_t::generate_shard_interval_result(void *shard_data, uint64_t interval_id)
+{
+    per_shard_t *per_shard = reinterpret_cast<per_shard_t *>(shard_data);
+    compute_shard_interval_result(per_shard, interval_id);
+    counters_t delta = per_shard->per_interval_delta.back();
+    counters_t cumulative = per_shard->per_interval_cumulative.back();
+    std::cerr << "[heartbeat] Interval " << interval_id << " in TID " << per_shard->tid
```

Comment:
This (and the one below) should probably be under a knob_verbose_ level.

Reply:
I thought for long traces it's useful to have periodic heartbeats that show that the progress.